1.29.0

matteobruni opened the pull request:
# 1.29.0

## New Features

- Added sizeOffset option to destroy split section
- Added inverse option to move gravity section
- Added fireworks preset package

## Bug Fixes

- Fixed move angle options
- Fixed move random and straight settings

dependabot[bot] committed:
build(deps): bump @angular/common from 12.0.4 to 12.0.5

Bumps [@angular/common](https://github.com/angular/angular/tree/HEAD/packages/common) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/common)

---
updated-dependencies:
- dependency-name: "@angular/common"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser-dynamic

Bumps [@angular/platform-browser-dynamic](https://github.com/angular/angular/tree/HEAD/packages/platform-browser-dynamic) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/platform-browser-dynamic)

---
updated-dependencies:
- dependency-name: "@angular/platform-browser-dynamic"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump riot from 5.4.4 to 5.4.5

Bumps [riot](https://github.com/riot/riot) from 5.4.4 to 5.4.5.
- [Release notes](https://github.com/riot/riot/releases)
- [Commits](https://github.com/riot/riot/compare/v5.4.4...v5.4.5)

---
updated-dependencies:
- dependency-name: riot
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 4.26.1 to 4.27.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 4.26.1 to 4.27.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.27.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/animations from 12.0.4 to 12.0.5

Bumps [@angular/animations](https://github.com/angular/angular/tree/HEAD/packages/animations) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/animations)

---
updated-dependencies:
- dependency-name: "@angular/animations"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1638 from matteobruni/dependabot/npm_and_yarn/dev/angular/common-12.0.5

build(deps): bump @angular/common from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1639 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-dynamic-12.0.5

build(deps): bump @angular/platform-browser-dynamic from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1640 from matteobruni/dependabot/npm_and_yarn/dev/riot-5.4.5

build(deps): bump riot from 5.4.4 to 5.4.5

matteobruni committed:
Merge pull request #1641 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/parser-4.27.0

build(deps-dev): bump @typescript-eslint/parser from 4.26.1 to 4.27.0

matteobruni committed:
Merge pull request #1642 from matteobruni/dependabot/npm_and_yarn/dev/angular/animations-12.0.5

build(deps): bump @angular/animations from 12.0.4 to 12.0.5

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-vue from 7.10.0 to 7.11.1

Bumps [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) from 7.10.0 to 7.11.1.
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v7.10.0...v7.11.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-vue
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser from 12.0.4 to 12.0.5

Bumps [@angular/platform-browser](https://github.com/angular/angular/tree/HEAD/packages/platform-browser) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/platform-browser)

---
updated-dependencies:
- dependency-name: "@angular/platform-browser"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.51.1 to 2.52.0

Bumps [rollup](https://github.com/rollup/rollup) from 2.51.1 to 2.52.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.51.1...v2.52.0)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.3.4 to 8.3.5

Bumps [postcss](https://github.com/postcss/postcss) from 8.3.4 to 8.3.5.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.3.4...8.3.5)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/forms from 12.0.4 to 12.0.5

Bumps [@angular/forms](https://github.com/angular/angular/tree/HEAD/packages/forms) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/forms)

---
updated-dependencies:
- dependency-name: "@angular/forms"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1643 from matteobruni/dependabot/npm_and_yarn/dev/eslint-plugin-vue-7.11.1

build(deps-dev): bump eslint-plugin-vue from 7.10.0 to 7.11.1

matteobruni committed:
Merge pull request #1644 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-12.0.5

build(deps): bump @angular/platform-browser from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1645 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.52.0

build(deps-dev): bump rollup from 2.51.1 to 2.52.0

matteobruni committed:
Merge pull request #1646 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.3.5

build(deps-dev): bump postcss from 8.3.4 to 8.3.5

dependabot[bot] committed:
build(deps): bump @angular/core from 12.0.4 to 12.0.5

Bumps [@angular/core](https://github.com/angular/angular/tree/HEAD/packages/core) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/core)

---
updated-dependencies:
- dependency-name: "@angular/core"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump typedoc from 0.21.0-beta.3 to 0.21.0-beta.4

Bumps [typedoc](https://github.com/TypeStrong/TypeDoc) from 0.21.0-beta.3 to 0.21.0-beta.4.
- [Release notes](https://github.com/TypeStrong/TypeDoc/releases)
- [Commits](https://github.com/TypeStrong/TypeDoc/commits)

---
updated-dependencies:
- dependency-name: typedoc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump sass-loader from 12.0.0 to 12.1.0

Bumps [sass-loader](https://github.com/webpack-contrib/sass-loader) from 12.0.0 to 12.1.0.
- [Release notes](https://github.com/webpack-contrib/sass-loader/releases)
- [Changelog](https://github.com/webpack-contrib/sass-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/sass-loader/compare/v12.0.0...v12.1.0)

---
updated-dependencies:
- dependency-name: sass-loader
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1647 from matteobruni/dependabot/npm_and_yarn/dev/angular/forms-12.0.5

build(deps): bump @angular/forms from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1648 from matteobruni/dependabot/npm_and_yarn/dev/angular/core-12.0.5

build(deps): bump @angular/core from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1649 from matteobruni/dependabot/npm_and_yarn/dev/typedoc-0.21.0-beta.4

build(deps-dev): bump typedoc from 0.21.0-beta.3 to 0.21.0-beta.4

matteobruni committed:
Merge pull request #1651 from matteobruni/dependabot/npm_and_yarn/dev/sass-loader-12.1.0

build(deps-dev): bump sass-loader from 12.0.0 to 12.1.0

dependabot[bot] committed:
build(deps-dev): bump @angular/compiler-cli from 12.0.4 to 12.0.5

Bumps [@angular/compiler-cli](https://github.com/angular/angular/tree/HEAD/packages/compiler-cli) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/compiler-cli)

---
updated-dependencies:
- dependency-name: "@angular/compiler-cli"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 4.26.1 to 4.27.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.27.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/router from 12.0.4 to 12.0.5

Bumps [@angular/router](https://github.com/angular/angular/tree/HEAD/packages/router) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/router)

---
updated-dependencies:
- dependency-name: "@angular/router"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1650 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/eslint-plugin-4.27.0

build(deps-dev): bump @typescript-eslint/eslint-plugin from 4.26.1 to 4.27.0

matteobruni committed:
Merge pull request #1653 from matteobruni/dependabot/npm_and_yarn/dev/angular/compiler-cli-12.0.5

build(deps-dev): bump @angular/compiler-cli from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1655 from matteobruni/dependabot/npm_and_yarn/dev/angular/router-12.0.5

build(deps): bump @angular/router from 12.0.4 to 12.0.5

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.52.0 to 2.52.1

Bumps [rollup](https://github.com/rollup/rollup) from 2.52.0 to 2.52.1.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.52.0...v2.52.1)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1654 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.52.1

build(deps-dev): bump rollup from 2.52.0 to 2.52.1

matteobruni committed:
feat: added sizeOffset option to destroy split section
feat: added inverse option to move gravity section
fix: fixed move angle options
fix: fixed move random and straight settings
build: added fireworks demo preset

matteobruni committed:
feat: added fireworks preset

dependabot[bot] committed:
build(deps-dev): bump @angular/cli from 12.0.4 to 12.0.5

Bumps [@angular/cli](https://github.com/angular/angular-cli) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v12.0.4...v12.0.5)

---
updated-dependencies:
- dependency-name: "@angular/cli"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/compiler from 12.0.4 to 12.0.5

Bumps [@angular/compiler](https://github.com/angular/angular/tree/HEAD/packages/compiler) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/compiler)

---
updated-dependencies:
- dependency-name: "@angular/compiler"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.39.0 to 5.39.1

Bumps [webpack](https://github.com/webpack/webpack) from 5.39.0 to 5.39.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.39.0...v5.39.1)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1658 from matteobruni/dependabot/npm_and_yarn/dev/angular/cli-12.0.5

build(deps-dev): bump @angular/cli from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1660 from matteobruni/dependabot/npm_and_yarn/dev/angular/compiler-12.0.5

build(deps): bump @angular/compiler from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1662 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.39.1

build(deps-dev): bump webpack from 5.39.0 to 5.39.1

dependabot[bot] committed:
build(deps-dev): bump @angular-devkit/build-angular

Bumps [@angular-devkit/build-angular](https://github.com/angular/angular-cli) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v12.0.4...v12.0.5)

---
updated-dependencies:
- dependency-name: "@angular-devkit/build-angular"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1661 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-angular-12.0.5

build(deps-dev): bump @angular-devkit/build-angular from 12.0.4 to 12.0.5

matteobruni committed:
feat: emitter rate delay and quantity are now RangeValues so they can be randomized

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
build: updated deps

matteobruni committed:
Merge pull request #1657 from matteobruni/dev

1.29.0

ghost commented:
## DeepCode's analysis on [#551940](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/8983cac130aa51874e0c5ec27845eb48ba0236f3/matteobruni/tsparticles/551940323d4d4f2619746d22426358837b569164/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) found:

- :warning: **1** warning :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">This endpoint handler performs a file system operation and does not use a rate-limiting mechanism. It may enable the attackers to perform Denial-of-service attacks. Consider using a rate-limiting middleware such as express-limit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/551940323d4d4f2619746d22426358837b569164/demo/vanilla/app.js#L42">app.js:42</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/551940323d4d4f2619746d22426358837b569164/demo/vanilla/app.js#L46">app.js:46</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/551940323d4d4f2619746d22426358837b569164/demo/vanilla/app.js#L50">app.js:50</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/8983cac130aa51874e0c5ec27845eb48ba0236f3/matteobruni/tsparticles/551940323d4d4f2619746d22426358837b569164/pr/_/%2Fdemo%2Fvanilla%2Fapp.js/javascript%2Fdc_interfile_project%2FNoRateLimitingForExpensiveWebOperation/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/8983cac130aa51874e0c5ec27845eb48ba0236f3/matteobruni/tsparticles/551940323d4d4f2619746d22426358837b569164/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

matteobruni committed:
build: updated manual demo preset, it checks also the speed

matteobruni committed:
chore(release): published new version

 - ng-particles@2.12.0
 - inferno-particles@1.12.0
 - jquery-particles@1.29.0
 - preact-particles@1.29.0
 - react-tsparticles@1.29.0
 - riot-particles@1.3.0
 - solid-particles@1.12.0
 - svelte-particles@1.12.0
 - particles.vue@2.12.0
 - particles.vue3@1.12.0
 - web-particles@1.2.0
 - tsparticles-editor@1.29.0
 - tsparticles@1.29.0
 - ng-particles-demo@1.14.0
 - editor-demo@1.14.0
 - inferno-particles-demo@1.12.0
 - jquery-particles-demo@1.14.0
 - preact-particles-demo@1.14.0
 - react-tsparticles-demo@1.14.0
 - riot-particles-demo@1.3.0
 - solid-particles-demo@1.12.0
 - svelte-demo@1.12.0
 - tsparticles-demo@1.14.0
 - particles.vue-demo@1.14.0
 - particles.vue3-demo@1.12.0
 - web-particles-demo@1.14.0
 - tsparticles-preset-big-circles@1.14.0
 - tsparticles-preset-bubbles@1.14.0
 - tsparticles-preset-confetti@1.14.0
 - tsparticles-preset-fire@1.14.0
 - tsparticles-preset-firefly@1.14.0
 - tsparticles-preset-fireworks@1.14.0
 - tsparticles-preset-fountain@1.14.0
 - tsparticles-preset-links@1.14.0
 - tsparticles-preset-sea-anemone@1.14.0
 - tsparticles-preset-snow@1.14.0
 - tsparticles-preset-stars@1.14.0
 - tsparticles-shape-bubble@1.14.0
 - tsparticles-shape-confetti@1.14.0
 - tsparticles-shape-heart@1.14.0
 - tsparticles-shape-multiline-text@1.14.0
 - tsparticles-shape-rounded-rect@1.14.0
 - tsparticles-shape-spiral@1.14.0
 - cra-template-particles-typescript@1.12.0
 - cra-template-particles@1.12.0
 - tsparticles-website@1.29.0

matteobruni committed:
build: website updated

matteobruni committed:
Merge pull request #1664 from matteobruni/dev

Release update
